src/types/resolveDescriptors.ts

```diff
@@ -844,7 +844,7 @@ export function resolveDescriptors(ctx: CompilerContext, Ast: FactoryAst) {
         for (const param of params) {
             if (isSelfId(param.name)) {
                 throwCompilationError(
-                    'Parameter name "self" is reserved',
+                    'Parameter name "self" is reserved, consider using "extends" function modifier',
```

Comment:
```suggestion
                    'Parameter name "self" is reserved for functions with "extends" modifier',
```

Same idea, but now we make it clear that only functions with "extends" modifier, i.e. extension functions, can use `self` as the name of the first parameter

Reply:
Have a look please!